### PR TITLE
feature/itx5other-740-get-products: created products page/module

### PR DESCRIPTION
**Pull request by OsipenkoDenis**
Requirement - created products page/login.
Changed approach to work on subscription when component is destroyed.

### CODE REVIEW

File: src/app/auth/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+export class AuthGuard implements CanActivate {
+
+  constructor(private auth: AuthService,
+              private router: Router) {
+  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+    if (this.auth.isAuthenticated()) {
```
Comment (romaaxa) [inline src/app/auth/auth.guard.ts:18]: `if (!this.auth.isAuthenticated())` and else is not needed then just return true

File: src/app/pages/home-page/home-page-smart.component.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 
 @Component({
   selector: 'home-page-smart',
-  template: `
-    <home-page-presentation
-            (onlogout)='logout($event)'
-            [isAuthenticated]='auth.isAuthenticated()'>
+  template: `<home-page-presentation
+      (onlogout)='logout($event)'
+      [isAuthenticated]='auth.isAuthenticated()'>
```
Comment (romaaxa) [inline src/app/pages/home-page/home-page-smart.component.ts:9]: inputs should be placed above outputs

File: src/app/pages/home-page/home-page.component.html [hunk 3]
```diff
@@ -21,7 +21,7 @@ <h3 class='home-page-header' routerLink='/'>
     </ul>
   </ng-template>
 </nav>
-<div class='home-page-container'>
+<div [class.home-page-container]='!isAuthenticated'>
```
Comment (romaaxa) [inline src/app/pages/home-page/home-page.component.html:24]: use ngClass

File: src/app/pages/home-page/home-page-smart.component.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
+import {AuthService} from '../../auth/services/auth.service';
 
 @Component({
   selector: 'home-page-smart',
-  template: `
-    <home-page-presentation
-            (onlogout)='logout($event)'
-            [isAuthenticated]='auth.isAuthenticated()'>
+  template: `<home-page-presentation
+      (onlogout)='logout($event)'
```
Comment (romaaxa) [inline src/app/pages/home-page/home-page-smart.component.ts:8]: method shoyld be called onLogout and output can be just logout

File: src/styles.scss [hunk 5]
```diff
@@ -196,10 +192,6 @@ img {
   border: #ccc solid 1px;
 }
 
-.round-img {
-  border-radius: 50%;
-}
-
 .register-page-form-control, .login-page-form-control {
   display: block;
   color: #888;
```
Comment (romaaxa) [inline src/styles.scss:197]: define color variables scss in common module and move all the colors into it. Use this link to find corresponding color names https://www.color-name.com/

File: src/app/auth/auth.guard.ts (hunk 1 above)
Comment (OsipenkoDenis) [reply src/app/auth/auth.guard.ts:18]: Done

File: src/app/pages/home-page/home-page-smart.component.ts (hunk 2 above)
Comment (OsipenkoDenis) [reply src/app/pages/home-page/home-page-smart.component.ts:9]: Done

File: src/app/pages/home-page/home-page-smart.component.ts (hunk 4 above)
Comment (OsipenkoDenis) [reply src/app/pages/home-page/home-page-smart.component.ts:8]: Done

File: src/styles.scss (hunk 5 above)
Comment (OsipenkoDenis) [reply src/styles.scss:197]: Done

**Pull request merged**